js/utils/helpers.js

```diff
@@ -1867,12 +1870,22 @@
                                     iframeClass = attr[1];
                                 } else if (attr[0] == "iframe-style") {
                                     iframeStyle = attr[1];
+                                } else if (attr[0] == "class") {
+                                    newClass += " " + attr[1];
                                 } else {
                                     iframeHTML +=  attr[0] + '="' + attr[1] + '"';
                                 }
                                 iframeHTML += " ";
                             });
-                            html += iframeHTML + "></iframe>";
+
+                            //During print we need to display that the iframe with YouTube video is replaced with a note
+                            if(isYTlink)
```

Comment:
As I mentioned in one of my comments, you can define extra classes for iframe, so computed iframe can have extra class. Therefore the `newClass` attribute that you added is something that you always have to add to your `iframeHTML`, so

```javascript
html += iframeHTML + newClass + "></iframe>";
if(isYTlink) {
  html = '<span class="video-info-in-print" style="visibility:hidden">' + videoText + "</span>" + html;
}
```

You should modify your existing youtube test case to include classes as well. So something like this

```
::: iframe [SOME LINK CAPTION](https://www.youtube.com/embed/op1-Cw_l1Ow){width=640 height=480 link=https://www.youtube.com/embed/op1-Cw_l1Ow .some-class .another-class} \n:::
```

Notice the two classes that I've added at the end.

Reply:
Also, try to follow the code styles that you see on the file. So the curly braces that you're adding after `if` should be on the same line (just like the rest of the code). 

And avoid defining `if` or `else` blocks without curly braces as well. Even if it's one line, you should add curly braces.

Reply:
The default behavior of the iframe is that, they shouldn't be hidden during print. Only when the user wants to, they can define the class - "hide-in-print" and make a decision to hide it. However, iframes containing YouTube videos must be hidden by default. Since I am using the same class to hide these, I am choosing to define them only if i`sYTLink` is true. 

Earlier, this code for adding the new classes was not present. So I guess, with my new code it will take care of that. I have changed this to suit the condition mentioned.